watertap/unit_models/ion_exchange_0D.py

```diff
@@ -599,539 +650,668 @@ def build(self):
-            units=pyunits.kg / pyunits.m**3,
-            bounds=(0, None),  # Perry's
-            doc="Regenerant dose required for regeneration per volume of resin [kg regenerant/m3 resin]",
-        )
+            self.langmuir = Var(
+                self.target_ion_set,
+                initialize=0.5,  # La < 1 is favorable isotherm
+                bounds=(0, 1.1),
+                units=pyunits.dimensionless,
+                doc="Langmuir isotherm coefficient",
+            )
 
-        self.t_regen = Var(
-            initialize=1800,
-            units=pyunits.s,
-            doc="Regeneration time",
-        )
+            self.mass_removed = Var(
+                self.target_ion_set,
+                initialize=1e6,
+                bounds=(0, None),
+                units=pyunits.mol,
+                doc="Sorbed mass of ion",
+            )
 
-        # ====== Backwashing ====== #
+            self.num_transfer_units = Var(
+                initialize=1e6,
+                bounds=(0, None),
+                units=pyunits.dimensionless,
+                doc="Number of transfer units",
+            )
 
-        self.bw_rate = Var(
-            initialize=5,
-            units=pyunits.m / pyunits.hour,
-            bounds=(4.5, 6.5),
-            doc="Backwash loading rate [m/hr]",
-        )
+            self.dimensionless_time = Var(
+                initialize=1,
+                units=pyunits.dimensionless,
+                doc="Dimensionless time",
+            )
 
-        self.bw_flow = Var(
-            initialize=0.1,
-            bounds=(0, None),
-            units=pyunits.m**3 / pyunits.s,
-            doc="Backwashing volumetric flow rate",
-        )
+            self.partition_ratio = Var(
+                initialize=100,
+                bounds=(0, None),
+                units=pyunits.dimensionless,
+                doc="Partition ratio",
+            )
 
-        self.t_bw = Var(
-            initialize=600, units=pyunits.s, bounds=(300, 1200), doc="Backwash time"
-        )
+            self.fluid_mass_transfer_coeff = Var(
+                self.target_ion_set,
+                initialize=1e-3,
+                bounds=(0, None),
+                units=pyunits.m / pyunits.s,
+                doc="Fluid mass transfer coefficient",
+            )
 
-        self.bed_expansion_frac = Var(
-            initialize=0.5,
-            bounds=(0.4, 1.2),
-            units=pyunits.dimensionless,
-            doc="Fraction of bed depth increase during backwashing",
-        )
+        if self.config.isotherm == IsothermType.freundlich:
 
-        self.bed_expansion_h = Var(
-            initialize=0.5,
-            units=pyunits.m,
-            bounds=(0, None),
-            doc="Additional column sidewall height required for bed expansion",
-        )
+            self.num_traps = 5  # TODO: make CONFIG option
+            self.trap_disc = range(self.num_traps + 1)
+            self.trap_index = self.trap_disc[1:]
```

Comment:
We should probably also try to standardize our use of the trapezoid rule for use in adsorption breakthroughs referencing the trapezoid rule implementation in the DAE package (I think), not for this PR though.